### DISCUSSION TITLE
pag sdk+qt相关问题

### DISCUSSION THREAD

**Discussion by qingpingwang** [Tencent/libpag#1692] [General] [answered]
![image](https://github.com/Tencent/libpag/assets/50129609/6e55d68f-d589-490c-b4c6-8d86f0d40b35)
在官方的qt案例里，已经绘制成功。但是pagsurface读取内容时出错。
想问一下官方有没有QQuickWidget相关案例，因为我想要qtwidgets

**Comment by kevingpqi123** [comment, marked as answer]
如果绘制成功 PAGSurface 读取内容出错，那就是读取的时候用法的问题，为啥要在 conSizeChanged 方法中读取呢，这里渲染尺寸已经发生了变化，当前帧没有渲染，这个时候读取肯定会有问题，你可以在外部单独调用 setProgress， flush ，然后读取当前帧数据

目前 QT 的渲染环境基于的是 QML， 如果需要支持 QWidgets，可以参考 QT 的相关封装自己实现下，pag 主体的渲染都在 C++ 层，平台侧仅仅实现接口封装和渲染环境提供
